proposals/0304-structured-concurrency.md

```diff
@@ -717,119 +765,238 @@ The isCancelled function is implemented as:
+
+> Note: Sadly it is not presently possible to implement this throwing/non-throwing functionality with a single function. The complex relationship of throwing `group.spawn` with a throwing `next` as well as coresponding throwing/non-throwing `AsyncSequence` conformances make it impossible to implement all in one function/type today.
+
+Note also that the `withThrowingTaskGroup` uses a `ThrowingTaskGroup<ChildTaskResult, Error>`, however specifying the type of that error is not possible. This is because this Failure parameter on the `ThrowingTaskGroup` in only used as future-proof API in case Swift were to gain typed throwing at some point in time. This design makes no promises nor does it assume typed throws are actually going to happen though.
 
 A task group _guarantees_ that it will `await` all tasks that were added to it before it returns.
 
 This waiting can be performed either: 
 - by the code within the task group itself (e.g., using `next()` repeatedly until it returns `nil`, described below), or
 - implicitly in the task group itself when returning from the `body`.
 
-By default, the task group will schedule child tasks added to the group on the default global concurrent executor. The `startingChildTasksOn` argument can
-be provided to override this behavior.
+By default, the task group will schedule child tasks added to the group on the default global concurrent executor. In the future is is likely that it will be possible to customize the executor tasks are started on with an optional executor parameter to `spawn`, e.g. like `startingChildTasksOn:`.
 
-##### Adding tasks to a group
+##### Spawning TaskGroup child tasks
 
-Within the `body` function, tasks may be added dynamically with the `add` operation. Each task produces a value of the same type (the `ResultType` generic parameter):
+Within the `body` function, tasks may be added dynamically with the spawn operation. Each task produces a value of the same type (the `ResultType` generic parameter):
 
 ```swift
-extension Task.Group { 
+extension TaskGroup {
   /// Add a child task to the group.
   ///
-  /// Returns true if the task was successfully added, false otherwise.
-  mutating func add(
+  /// Returns true if the task was successfully spawned, false otherwise.
+  /// A cancelled group will reject any further additions of tasks into it.
+  mutating func spawn(
       overridingPriority: Priority? = nil,
-      operation: @concurrent @escaping () async throws -> TaskResult
-  ) async -> Bool
+      operation: @concurrent @escaping () async -> TaskResult
+  ) -> Self.Spawned
+  
+  public struct Spawned: Sendable {
+    /// Returns `true` if the task was successfully spawned in the task group,
+    /// `false` otherwise which means that the group was already cancelled and
+    /// refused to accept spawn a new child task.
+    public var successfully: Bool 
```

Comment:
(This of course is just `var sucessfully: Bool { handle != nil }`  so does not need storage)